Mage.Sets/src/mage/cards/c/CreepingInn.java

```diff
@@ -0,0 +1,110 @@
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.filter.StaticFilters;
+import mage.game.ExileZone;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetCardInGraveyard;
+import mage.util.CardUtil;
+
+import java.util.UUID;
+
+/**
+ * @author LePwnerer
+ */
+public final class CreepingInn extends CardImpl {
+
+    public CreepingInn(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT, CardType.CREATURE}, "");
+
+        this.subtype.add(SubType.HORROR);
+        this.subtype.add(SubType.CONSTRUCT);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(7);
+        this.nightCard = true;
+
+        // Whenever Creeping Inn attacks, you may exile a creature card from your graveyard.
+        // If you do, each opponent loses X life and you gain X life,
+        // where X is the number of creature cards exiled with Creeping Inn.
+        this.addAbility(new AttacksTriggeredAbility(new CreepingInnEffect()));
+
+        // {4}: Creeping Inn phases out.
+        this.addAbility(new SimpleActivatedAbility(Zone.BATTLEFIELD, new PhaseOutSourceEffect(), new ManaCostsImpl("{4}")));
+    }
+
+    private CreepingInn(final CreepingInn card) {
+        super(card);
+    }
+
+    @Override
+    public CreepingInn copy() {
+        return new CreepingInn(this);
+    }
+}
+
+class CreepingInnEffect extends OneShotEffect {
+
+    public CreepingInnEffect() {
+        super(Outcome.Exile);
+        this.staticText = "you may exile a creature card from your graveyard. " +
+                "If you do, each opponent loses X life and you gain X life, " +
+                "where X is the number of creature cards exiled with Creeping Inn.";
+    }
+
+    public CreepingInnEffect(final CreepingInnEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public CreepingInnEffect copy() {
+        return new CreepingInnEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        Permanent permanent = game.getPermanent(source.getSourceId());
+        game.informPlayers(permanent.getName());
+        if (permanent != null && player != null) {
+            UUID exileId = CardUtil.getExileZoneId(source.getSourceId().toString() + permanent.getZoneChangeCounter(game), game);
+            TargetCardInGraveyard target = new TargetCardInGraveyard(StaticFilters.FILTER_CARD_CREATURE_YOUR_GRAVEYARD);
+            target.setNotTarget(true);
+            if (target.canChoose(source.getSourceId(), player.getId(), game) && player.chooseUse(outcome, "Exile a creature card from your graveyard?", source, game)) {
+                if (player.choose(Outcome.Exile, target, source.getId(), game)) {
```

Comment:
you can shortcut most of this by using `minNumTargets` set to 0, so you don't have to check for eligible targets or ask yes or no